astropy/table/column.py

```diff
@@ -1407,6 +1407,30 @@ def insert(self, obj, values, axis=0):
             A copy of column with ``values`` and ``mask`` inserted.  Note that the
             insertion does not occur in-place: a new column is returned.
         """
+
+        def warn_quantities(values):
```

Comment:
note: I've put this in a closure mostly to avoid polluting the scope of the parent function, but this could also be achieved using `del`. Anyhow, this is an implementation detail, we can change it.